src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+        if (CollectionUtils.isNotEmpty(unprocessedExternalObjectDirectoryEntities)) {
+            externalObjectDirectoryService.updateStatus(
+                EodHelper.armDropZoneStatus(),
+                userAccount,
+                unprocessedExternalObjectDirectoryEntities.stream().map(ExternalObjectDirectoryEntity::getId).toList(),
+                timeHelper.currentOffsetDateTime());
+            for (ExternalObjectDirectoryEntity unprocessedExternalObjectDirectoryEntity : unprocessedExternalObjectDirectoryEntities) {
+                log.warn("Unable to process ARM responses for EOD {}", unprocessedExternalObjectDirectoryEntity.getId());
+            }
+        }
+    }
+
+    private void processResponseFileByHashcode(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor, String manifestName) {
+        try {
+            List<String> responseFiles = armDataManagementApi.listResponseBlobs(batchUploadFileFilenameProcessor.getHashcode());
+            if (CollectionUtils.isNotEmpty(responseFiles)) {
+                ResponseFilenames responseFilenames = getArmResponseFilenames(responseFiles, manifestName);
+                ArmBatchResponses armBatchResponses = new ArmBatchResponses();
+                if (responseFilenames.containsResponses()) {
+                    //Put response files into their respective groups based on the contents relation id (EOD id)
+                    processCreateRecordResponseFiles(responseFilenames.getCreateRecordResponses(), armBatchResponses);
+                    processUploadResponseFiles(responseFilenames.getUploadFileResponses(), armBatchResponses);
+                    processInvalidFiles(responseFilenames.getInvalidLineResponses(), armBatchResponses);
+                    //Process the final results
+                    processBatchResponseFiles(batchUploadFileFilenameProcessor, armBatchResponses);
+                }
+            } else {
+                log.info("Unable to find response files starting with {} for manifest {}", batchUploadFileFilenameProcessor.getHashcode(), manifestName);
+            }
+        } catch (Exception e) {
+            log.error("Unable to process responses for file {}", batchUploadFileFilenameProcessor.getBatchMetadataFilenameAndPath(), e);
+        }
+    }
+
+    private void processBatchResponseFiles(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor, ArmBatchResponses armBatchResponses) {
+        armBatchResponses.getArmBatchResponseMap().values().forEach(
+            armResponseBatchData -> {
+                if (nonNull(armResponseBatchData.getInvalidLineFileFilenameProcessor())
+                    && (nonNull(armResponseBatchData.getCreateRecordFilenameProcessor())
+                    || nonNull(armResponseBatchData.getArmResponseUploadFileRecord()))) {
+
+                    beforeProcessingResponseFiles(armResponseBatchData.getExternalObjectDirectoryId());
```

Comment:
author note: added `beforeProcessingResponseFiles` abstract method which has a no op implementation in `ArmBatchProcessResponseFilesImpl`